miniwin/src/internal/d3drmobject_impl.h

```diff
@@ -7,6 +7,11 @@
 
 template <typename T>
 struct Direct3DRMObjectBaseImpl : public T {
+	Direct3DRMObjectBaseImpl(const Direct3DRMObjectBaseImpl& other) : T(other)
+	{
+		// We should not call the callback twice
+		m_callbacks.clear();
```

Comment:
This won't be necessary. Since you overrode the copy constructor the default behavior of copying the members won't apply  anymore. So no need to clear callbacks since it's already empty.  All the other members won't be copied either m If that's desired you can have an empty body for this ctor

Reply:
just trying to follow @madebr's suggestion :shrug: